tests/utils.py

```diff
@@ -1018,7 +1018,12 @@ def snapshot_context(
         except Exception as e:
             pytest.fail("Could not connect to test agent: %s" % str(e), pytrace=False)
         else:
-            r = conn.getresponse()
+            r = None
+            while r is None:
```

Comment:
nit: Do we just want to let this hit the timeout on the job? My only thought here is not letting subsequent tests run because this just keeps trying, so we won't know all tests that are having issues until we get this test to pass.